### PR TITLE
[PipelineRun 1/ ][crag-core] PipelineRun -> DagsterRun, persist as PipelineRun in storage layer

### PR DESCRIPTION
**Pull request by dpeng817**
Switches over PipelineRun into DagsterRun (keeps around PipelineRun class so as to not overwhelm PR with rename stuff)

### CODE REVIEW

File: python_modules/dagster/dagster/core/storage/pipeline_run.py [hunk 1]
```diff
@@ -375,6 +397,14 @@ def tags_for_partition_set(partition_set, partition):
         return {PARTITION_NAME_TAG: partition.name, PARTITION_SET_TAG: partition_set.name}
 
 
+# previous name for NodeHandle was SolidHandle
```
Comment (prha) [inline python_modules/dagster/dagster/core/storage/pipeline_run.py:400]: update comment

File: python_modules/dagster/dagster/core/storage/pipeline_run.py [hunk 2]
```diff
@@ -375,6 +397,14 @@ def tags_for_partition_set(partition_set, partition):
...
+register_serdes_tuple_fallbacks({"PipelineRun": DagsterRun})
+
+
+class PipelineRun(DagsterRun):
```
Comment (prha) [inline python_modules/dagster/dagster/core/storage/pipeline_run.py:404]: add comment here about the purpose of this?

Comment (prha) [reply python_modules/dagster/dagster/core/storage/pipeline_run.py:404]: Do we need to worry about breaking user's code that are relying on types?

If we have something typed as `PipelineRun` and we return a `DagsterRun`, we'll get errors, right?

**Comment by dpeng817** [comment]
Changing the top-level class will break compat for people who may be creating their own pipeline runs, or using for type annotations. Abandoning until / if we decide we're comfortable breaking

**Pull request closed without merging**